tools/sync/projadm.py

```diff
@@ -168,13 +174,12 @@ def project_delete(doit, logger, project, messages):
         raise Exception("invalid call to project_delete(): missing project")
 
     logger.info("Deleting project {} and its index data".format(project))
-    exec_command(doit, logger,
-                 [messages, '-n', 'project', '-t', project, 'delete'],
-                 "deletion of the project failed")
 
-    src_root = exec_command(True, logger,
-                            [messages, '-n', 'config', '-t', 'get',
-                             'sourceRoot'], "cannot get config")
+    if doit:
+        delete_project(logger, project, host)
+
+    src_root = get_config_value(logger, 'sourceRoot', host)
```

Comment:
should probably check the return value afterwards given the function can return `None`

In general it is not clear to me what are the failure scenarios for `requests`, e.g. what if:
  - the URI is specified incorrectly
  - the HTTP request fails
  - the service returns error

Reply:
If the service returns `3xx`, `4xx`, `5xx` then `if r:` returns `False`.

On error `ConnectionError` is thrown.